sdv/metadata/multi_table.py

```diff
@@ -532,6 +533,79 @@ def validate(self):
                 'The metadata is not valid' + '\n'.join(str(e) for e in errors)
             )
 
+    def _validate_missing_tables(self, data):
+        errors = []
+        missing_tables = set(self.tables) - set(data)
+        if missing_tables:
+            errors.append(f'The provided data is missing the tables {missing_tables}.')
+
+        return errors
+
+    def _validate_all_tables(self, data, table_synthesizers=None):
+        errors = []
+        for table_name, table_data in data.items():
+            try:
+                if table_synthesizers:
+                    table_synthesizers[table_name].validate(table_data)
```

Comment:
I think we should keep this out of the metadata. It's a strange cyclic relationship. I think we can just do the case in the else and then do the case in the if only in the `MultiTableSynthesizer`. That would also make it clear why `MultiTableMetadata.validate_data` isn't being called in the synthesizer.

Reply:
The issue of doing that is the code gets duplicated, which is why I took this approach. It is confusing though, so I guess I'll just leave the duplicated code there.